dask_jobqueue/deploy/cluster_manager.py

```diff
@@ -83,8 +84,14 @@ def __init__(self, adaptive_options={}):
         self._adaptive_options = adaptive_options
         self._adaptive_options.setdefault('worker_key', self.worker_key)
 
+    def sleep_until_n_workers(self, n):
+        '''Block by sleeping until we have n active workers'''
+        while self._count_active_workers() < n:
```

Comment:
In ClusterManager, we should not rely on functions from JobQueueCluster. You need to find something than can be called on the scheduler remotly at one point. Or we need the ClusterManager to be informed of any incoming worker.
What shall be kept in mind is that the ClusterManager is potentially another process than the Scheduler, and that it is build to be usable by other Cluster implementation than those of dask-jobqueue.